tests/test_cognitoidp/test_cognitoidp.py

```diff
@@ -1145,7 +1295,67 @@ def test_admin_get_user():
 
     result = conn.admin_get_user(UserPoolId=user_pool_id, Username=username)
     result["Username"].should.equal(username)
-    result["UserAttributes"].should.have.length_of(1)
+    result["UserAttributes"].should.have.length_of(2)
+
+
+@mock_cognitoidp
+def test_admin_get_user_with_username_attributes():
+    conn = boto3.client("cognito-idp", "us-west-2")
+
+    username = "test@example.com"
+    value = str(uuid.uuid4())
+    user_pool_id = conn.create_user_pool(
+        PoolName=str(uuid.uuid4()), UsernameAttributes=["email", "phone_number"]
+    )["UserPool"]["Id"]
+    conn.admin_create_user(
+        UserPoolId=user_pool_id,
+        Username=username,
+        UserAttributes=[
+            {"Name": "thing", "Value": value},
+            {"Name": "phone_number", "Value": "+123456789"},
+        ],
+    )
+    # verify user can be queried by email
+    result = conn.admin_get_user(UserPoolId=user_pool_id, Username=username)
+    result["Username"].should_not.equal(username)
+    result["UserAttributes"].should.have.length_of(4)
+
+    def _verify_attribute(name, v):
+        attr = [a for a in result["UserAttributes"] if a["Name"] == name]
+        attr.should.have.length_of(1)
+        attr[0]["Value"].should.equal(v)
+
+    _verify_attribute("phone_number", "+123456789")
+    _verify_attribute("email", "test@example.com")
+
+    # verify user can be queried by phone number
+    result = conn.admin_get_user(UserPoolId=user_pool_id, Username="+123456789")
+
+    result["Username"].should_not.equal(username)
+    result["UserAttributes"].should.have.length_of(4)
+    _verify_attribute("phone_number", "+123456789")
+    _verify_attribute("email", "test@example.com")
+
+    # verify that the generate user sub is a valid UUID v4
+    [user_sub] = [
+        attr["Value"] for attr in result["UserAttributes"] if attr["Name"] == "sub"
+    ]
+    user_sub_valid = False
+    try:
```

Comment:
This shouldn't need a try-catch - we can just call `uuid.UUID(user_sub)` as is, and let it fail if it's invalid